### PR TITLE
ROX-13421: Enable roxctl netpol generate and add tech-preview messages

### PR DESCRIPTION
**Pull request by fredrb**
## Description

This PR enables the `generate netpol` in `roxctl`, adds tech preview text to help messages and a warning to `stderr` when the command is used. 

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] ~~Determined and documented upgrade steps~~
- [ ] ~~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

Documentation is already being handled, as per comment below. 

## Testing Performed

`roxctl --help`
```
Usage:
  ./bin/darwin_amd64/roxctl [command]

Available Commands:
   ...
  generate    (Technical Preview) Commands related to generating different resources.
...
```

`roxctl generate --help`
```
Commands related to generating different resources.

** This is a Technology Preview feature **
Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.For more information about the support scope of Red Hat Technology Preview features, see https://access.redhat.com/support/offerings/techpreview/.

Usage:
  ./bin/darwin_amd64/roxctl generate [command]

Available Commands:
  netpol      Recommend Network Policies based on deployment information.
...
```

`roxctl generate netpol --help`
```
Based on a given folder containing deployment YAMLs, will generate a list of recommended Network Policies. Will write to stdout if no output flags are provided.

** This is a Technology Preview feature **
Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.For more information about the support scope of Red Hat Technology Preview features, see https://access.redhat.com/support/offerings/techpreview/.

Usage:
  ./bin/darwin_amd64/roxctl generate netpol <folder-path> [flags]
...
```

`roxctl generate netpol ./sample-folder`
```
WARN:	This is a Technology Preview feature. Red Hat does not recommend using them in production.
...
```

### CODE REVIEW

File: roxctl/help/help.properties
Comment (vikin91) [reply roxctl/help/help.properties:178]: Yes, I think so.

**Review by vikin91 — COMMENTED** [review]
The code looks great! Could you maybe also update:
- this [readme](https://github.com/stackrox/stackrox/blob/7828a9f9a200126d3d06592b7172405f921f53ad/roxctl/generate/README.md)
- Bats tests for this feature `local-roxctl-tests`

**Comment by Maddosaurus** [comment]
+1 for the bats tests and the README.
Also, leaving this for tracking purposes: The associated PM will take care of updating and extending our product documentation with regards to this feature.
https://srox.slack.com/archives/C02HX541YQN/p1667577408910199?thread_ts=1667570132.941969&cid=C02HX541YQN

**Review by Maddosaurus — COMMENTED** [review]
+1 for the bats tests and the README.
Also, leaving this for tracking purposes: The associated PM will take care of updating and extending our product documentation with regards to this feature.
https://srox.slack.com/archives/C02HX541YQN/p1667577408910199?thread_ts=1667570132.941969&cid=C02HX541YQN

**Review by janisz — APPROVED** [review]

**Comment by fredrb** [comment]
@vikin91 

> Bats tests for this feature local-roxctl-tests

They seem to failing because the feature is now enabled, and the `*release` tests expect a `unknown command "generate"` message.

Any suggestions on what should be tested on roxctl release? The behavior should be the same as in the dev version, right? Would you be ok in removing the release test?

**Comment by vikin91** [comment]
Yes release tests should work now identically to the development tests. I would propose to do the following:
- remove the FF parts from the tests completely
- copy dev tests to the release (removing the `unknown command "generate"` assert) and see if they work: 
    - if yes keep them && merge 
    - if not, then we should give it a quick look why they fail (maybe they contain a reason to hold releasing this feat?)

File: roxctl/help/help.properties
Comment (msugakov) [inline roxctl/help/help.properties:176]: There's a missing space. Also, the trailing dot becomes a part of URL, I think we should safely get rid of it.
```suggestion
generate.long = Commands related to generating different resources.\n\n** This is a Technology Preview feature **\nTechnology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process. For more information about the support scope of Red Hat Technology Preview features, see https://access.redhat.com/support/offerings/techpreview/
```

File: roxctl/help/help.properties
Comment (msugakov) [inline roxctl/help/help.properties:179]:
```suggestion
generate.netpol.long = Based on a given folder containing deployment YAMLs, will generate a list of recommended Network Policies. Will write to stdout if no output flags are provided.\n\n** This is a Technology Preview feature **\nTechnology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process. For more information about the support scope of Red Hat Technology Preview features, see https://access.redhat.com/support/offerings/techpreview/
```

**Review by msugakov — APPROVED** [review]
Feel free to ship it after addressing comments.

File: tests/roxctl/bats-tests/local/roxctl-netpol-generate-release.bats
Comment (msugakov) [inline tests/roxctl/bats-tests/local/roxctl-netpol-generate-release.bats:3]: [not blocking the merge]
Is this file a copy-paste of `roxctl-netpol-generate-development.bats` with the only difference that `roxctl-development` is changed for `roxctl-release`? I don't think that massively copy-pasting anything is healthy practice that we can sustainably do. If there's no way to parameterize tests that they run development and release builds, I would vote to keep only the test that checks `roxctl-release`.

**Review by vikin91 — APPROVED** [review]

File: tests/roxctl/bats-tests/local/roxctl-netpol-generate-release.bats
Comment (fredrb) [reply tests/roxctl/bats-tests/local/roxctl-netpol-generate-release.bats:3]: Wouldn't it work to parameterize the specific parts with env vars and `source generic_test.sh` from two bats tests files? (didn't test)

Comment (vikin91) [reply tests/roxctl/bats-tests/local/roxctl-netpol-generate-release.bats:3]: No, unfortunately sourcing would not work. There was a thread about this somewhere in the Bats GH/Wiki. Maybe I will give it a look if they got any new feats regarding this.

**Pull request merged**
